### ISSUE TITLE
[no-duplicates] Bad auto fix for typescript

### ISSUE THREAD

**Issue by latin-1** [import-js/eslint-plugin-import#2697]
Regardless of the `prefer-inline` option.

### Input

```ts
import type { A } from "./mod";
import { type B, C } from "./mod";
```

### Output

```ts
import type { A, type B, C } from "./mod";
```

cc @snewcomer

**Comment by snewcomer** [comment]
@latin-1 Thanks for the issue!  Yes, it doesn't look like it is performing correctly.  Both the type and inline type specifiers for an import statement are not allowed.  Looking at this and a few other issues today.

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBDAnmApnA3nAggGgctAITgF84AzKCEOAcgDoB6ECAE1oG4g

If `prefer-inline: true`, I'm guessing it should look like this

```
import { type A, type B, C } from "./mod";
```

If it is false or not specified, it should stay the same.

```
import type { A } from "./mod";
import { type B, C } from "./mod";
```

**Comment by snewcomer** [comment]
Furthermore, I wonder if instead we should make our best effort to collapse.  

prefer-inline false
```
import type { A, B } from "./mod";
import { C } from "./mod";
```

The easier case to handle is prefer-inline true.  However, if false, this isn’t terribly hard to handle as well.  Basically we pull all inline types and type imports from other Type ImportDeclarations to the first Type ImportDeclaration and leave the Value ImportDeclarations to their own import statement.  The flexibility here is important esp when considering many (>2) duplicate import statements.

**Comment by ljharb** [comment]
Yes, I think it would be ideal to maximally collapse, preserving order as best as can be (`import/order` can clean that up further).

**Comment by dylang** [comment]
Possibly related: https://github.com/import-js/eslint-plugin-import/issues/2792
